src/modules/package.rs

```diff
@@ -0,0 +1,89 @@
+use super::Segment;
+use crate::context::Context;
+use ansi_term::Color;
+use serde_json;
+use std::fs::File;
+use std::io::Read;
+use std::path::PathBuf;
+use toml;
+
+/// Creates a segment with the current package version
+pub fn segment(context: &Context) -> Option<Segment> {
+    match get_package_version(context) {
+        Some(package_version) => {
+            const RUST_CHAR: &str = "📦";
+            const SEGMENT_COLOR: Color = Color::Red;
+
+            let mut segment = Segment::new("package");
+            // TODO: Should have prefix "is "
+            segment.set_style(SEGMENT_COLOR);
+
+            segment.set_value(format!("{} {}", RUST_CHAR, package_version));
+
+            Some(segment)
+        }
+        None => None,
+    }
+}
+
+fn has_rs_files(dir_entry: &PathBuf) -> bool {
+    let is_rs_file =
+        |d: &PathBuf| -> bool { d.is_file() && d.extension().unwrap_or_default() == "rs" };
+    let is_cargo_toml =
+        |d: &PathBuf| -> bool { d.is_file() && d.file_name().unwrap_or_default() == "Cargo.toml" };
+
+    is_rs_file(&dir_entry) || is_cargo_toml(&dir_entry)
+}
+
+fn has_js_files(dir_entry: &PathBuf) -> bool {
+    let is_js_file =
+        |d: &PathBuf| -> bool { d.is_file() && d.extension().unwrap_or_default() == "js" };
+    let is_node_modules =
+        |d: &PathBuf| -> bool { d.is_dir() && d.file_name().unwrap_or_default() == "node_modules" };
+    let is_package_json = |d: &PathBuf| -> bool {
+        d.is_file() && d.file_name().unwrap_or_default() == "package.json"
+    };
+
+    is_js_file(&dir_entry) || is_node_modules(&dir_entry) || is_package_json(&dir_entry)
+}
+
+fn get_package_version(context: &Context) -> Option<String> {
+    let is_rs_project = context.dir_files.iter().any(has_rs_files);
+    if !is_rs_project {
+        let is_js_project = context.dir_files.iter().any(has_js_files);
+        if !is_js_project {
+            return None;
+        } else {
+            let mut file = File::open("package.json").unwrap();
```

Comment:
This could possibly throw if you don't have read permissions on `package.json`.
`unwrap()` would cause the prompt to panic in that case.

Reply:
@matchai So remove the `.unwrap()`?

Reply:
We should handle the error instead of letting it panic.
I'd take a look at Rust error handling solutions.